test/test_circularbuffer.jl

```diff
@@ -0,0 +1,34 @@
+using DataStructures
+using Base.Test
+
+
+cb = CircularBuffer{Int}(5)
+@test length(cb) == 0
+@test capacity(cb) == 5
+@test_throws ErrorException first(cb)
```

Comment:
It looks like this is throwing an ArgumentError instead of an ErrorException on 0.4+. Maybe this has to be `@test_throws Union{ErrorException,ArgumentError} first(cb)`